brayns/engine/EngineObjectFactory.h

```diff
@@ -101,73 +100,28 @@ class EngineObjectFactory
 
         const auto &creator = *it;
 
-        return creator->create(data);
-    }
-
-    template<typename SubT>
-    void registerType()
-    {
-        auto temp = std::make_unique<SubT>();
-        auto name = temp->getName();
-        _items.push_back(std::make_unique<FactoryEntry<SubT>>(name));
+        return creator->create(payload.params);
     }
 
-private:
-    std::vector<std::unique_ptr<IFactoryEntry>> _items;
-};
-
-template<typename T>
-class GenericObject
-{
-public:
-    GenericObject() = default;
-    GenericObject(T &object, EngineObjectFactory<T> &factory)
-        : _systemObject(&object)
-        , _factory(&factory)
-        , _value(Json::parse("{}"))
+    std::unique_ptr<T> createOr(const EngineObjectData &payload, const T &defaultValue) const
     {
-    }
-
-    void setName(const std::string &name)
-    {
-        _name = name;
-    }
+        auto &name = payload.name;
+        if (name.empty())
+        {
+            return defaultValue.clone();
+        }
 
-    void setValue(const JsonValue &value)
-    {
-        _value = value;
+        return create(payload);
     }
 
-    std::unique_ptr<T> create()
+    template<typename SubT>
+    void registerType()
     {
-        if (_name.empty())
-        {
-            return _systemObject->clone();
-        }
-
-        return _factory->create(_name, _value);
+        auto name = SubT().getName();
```

Comment:
getName() could be static maybe ?

Reply:
I though about an inline static variable, but I didnt implement it because the getName() is needed when calling get-camera-type, where you only have access to the abstract type.